rustls/src/client/mod.rs

```diff
@@ -172,6 +173,55 @@ impl ClientConfig {
     }
 }
 
+/// Encodes ways a client can know the expected name of the server.
+///
+/// This currently covers knowing the DNS name of the server, but
+/// will be extended in the future to knowing the IP address of the
+/// server, as well as supporting privacy-preserving names for the
+/// server ("ECH").  For this reason this enum is `non_exhaustive`.
+#[non_exhaustive]
+#[derive(Debug, PartialEq, Clone)]
+pub enum ServerName {
```

Comment:
fwiw, I don't like `ServerIdentity` much either, because of the overlap with PKI terms. Maybe just `Server`?

```
pub enum Server {
    /// The server is identified by a DNS name.  The name
    /// is sent in the TLS Server Name Indication (SNI)
    /// extension.
    DnsName(webpki::DnsName),
    EncryptedHello(...),
    IpAddress(...)
}
```

Reply:
i chose "name" here for consistency with the `subjectAlternativeName` X.509 extension, cos ultimately that's one important place where this data is used.

Reply:
But in the case of ECH, if I understand correctly, the variant would contain a bunch of other data apart from just the `subjectAlternativeName`, right?

Reply:
Correct, yes. But that data is still all related to server naming because protecting the confidentiality of SNI is the ultimate goal of ECH (even if the mechanism is much more general than ESNI was).